drivers/i3c/i3c_dw.c

```diff
@@ -2385,8 +2372,10 @@ static DEVICE_API(i3c, dw_i3c_api) = {
 	.attach_i3c_device = dw_i3c_attach_device,
 	.reattach_i3c_device = dw_i3c_reattach_device,
 	.detach_i3c_device = dw_i3c_detach_device,
-	.attach_i2c_device = dw_i3c_i2c_attach_device,
-	.detach_i2c_device = dw_i3c_i2c_detach_device,
+
+	/* We dynamically attach I2C devices on transfer. */
+	.attach_i2c_device = NULL,
+	.detach_i2c_device = NULL,
```

Comment:
It's not really necessary to set unused api functions to NULL. You can remove these lines

Reply:
I just wanted to to make it really clear why these APIs are not implemented, but fair enough.

Reply:
yeah... it may be helpful to add a comment block within that i2c transfer function where that dynamic writing of the registers is happening

Reply:
done